pkg/volume/emptydir/quota.go

```diff
@@ -123,16 +138,21 @@ func (xqa *xfsQuotaApplicator) Apply(dir string, medium api.StorageMedium, pod *
 
 func (xqa *xfsQuotaApplicator) applyQuota(volDevice string, quota resource.Quantity, fsGroupID int64) error {
 	_, stderr, err := xqa.cmdRunner.RunApplyQuotaCommand(volDevice, quota, fsGroupID)
-	if err != nil {
-		return err
-	}
+
 	// xfs_quota is very happy to fail but return a success code, likely due to its
 	// interactive shell approach. Grab stderr, if we see anything written to it we'll
 	// consider this an error.
+	//
+	// If we exit non-zero *and* write to stderr, stderr is likely to have the details on what
+	// actually went wrong, so we'll use this as the error message instead.
 	if len(stderr) > 0 {
 		return fmt.Errorf("xfs_quota wrote to stderr: %s", stderr)
```

Comment:
I would either just return stderr, or change the error text to something like `error applying quota: %s`.